### PR TITLE
Compare Labels working for user created private labels on search page DL-190

### PR DESCRIPTION
**Pull request by rcavezza**
Now can select your own private labels for comparison on search results page. 

Fixes bug for non logged in users on the search results page. 

Cleans up code for saved searches/compare labels and puts them into their own js files.

Demo video https://share.getcloudapp.com/v1uPnRKG

DL-190

Screenshot 2023-04-18 at 7 11 56 PM

Commit message below
Gets compare labels working with private created labels and also gets…
… saved search not erroring out w/ non logged in users plus some generic code cleanup

Screenshot 2023-04-25 at 6 16 00 PM
Note: Added above screenshot because we now show more data to show uniqueness per created label as per David's comments last week.

### CODE REVIEW

**Review by ColeDCrawford — APPROVED** [review]
Nice, this looks good @rcavezza . Could you fix the merge conflicts and then we can get this merged to `main`?

**Comment by rcavezza** [comment]
Got `pre-commit` to work but needed to make a change to precommit's yaml file for it to work on my machine. I tried to not add that file in, but then pre-commit wouldn't let me write a commit message 😂 😂

**Comment by ColeDCrawford** [comment]
I think that should be ok, `python3` is more permissive and seems fine. I think you can use `git commit --no-verify -m "commit message"` to skip a precommit in the future: https://stackoverflow.com/questions/7230820/skip-git-commit-hooks

File: dle/search/static/search/es_search.js [hunk 1]
```diff
@@ -6,6 +6,7 @@
 var globalSearchTerm = '';
 var queryType = 'match'; // knn, simpleQueryString, match
 
+var foundDrugLabels = [];
```
Comment (ColeDCrawford) [inline dle/search/static/search/es_search.js:9]: This might be extraneous?

**Pull request merged**